src/app/domain/geolocation/services/tracking.ts

```diff
@@ -43,13 +46,24 @@ export const forceUploadGeolocationTrackingData = async (): Promise<void> => {
 
 interface GeolocationTrackingStatus {
   enabled: boolean
+  quotaExceeded: boolean
+}
+
+export const isGeolocationTrackingEnabled = async (): Promise<boolean> => {
+  const status = await BackgroundGeolocation.getState()
+
+  return status.enabled
 }
 
-export const getGeolocationTrackingStatus =
-  async (): Promise<GeolocationTrackingStatus> => {
-    const status = await BackgroundGeolocation.getState()
+export const getGeolocationTrackingStatus = async (
+  client: CozyClient | undefined
+): Promise<GeolocationTrackingStatus> => {
+  if (!client) {
+    throw new Error('You must be logged in to use geolocation tracking feature')
```

Comment:
those kind of errors are not shown to the end user I think we shouldn't use a syntax that looks like end user style, it can give the impression we're supposed to show an error toaster

Reply:
Interesting, never thought about this. Actually it goes back to the intent call so it could be displayed. But here it is not. For photo backup, it is. I will do nothing for the moment but I will keep that in mind.